### ISSUE TITLE
Rails 4.2.2 compatibility

### ISSUE THREAD

**Issue by voxik** [thoughtbot/shoulda-matchers#743]
Testing against RoR 4.2.2, I observe following test errors:

```
  1) Shoulda::Matchers::ActiveModel::DisallowValueMatcher when the value is outside of the range of the column not qualified with strict accepts, failing with the correct message
     Failure/Error: expect(&assertion).to fail_with_message(message)
       Expectation should have failed with message:
         Did not expect errors when attr is set to 100000,
         got RangeError: "100000 is out of range for ActiveRecord::Type::Integer with limit 2"
       However, the expectation did not fail.
     # ./spec/unit/shoulda/matchers/active_model/disallow_value_matcher_spec.rb:93:in `block (4 levels) in <top (required)>'

... snip ...

  9) Shoulda::Matchers::ActiveModel::AllowValueMatcher when the value is outside of the range of the column qualified with strict qualified with a message ignores any specified message
     Failure/Error: expect(&assertion).to fail_with_message(message)
       Expectation should have failed with message:
         Did not expect exception to include "some message" when attr is set to 100000,
         got RangeError: "100000 is out of range for ActiveRecord::Type::Integer with limit 2"
       However, the expectation did not fail.
     # ./spec/unit/shoulda/matchers/active_model/allow_value_matcher_spec.rb:297:in `block (5 levels) in <top (required)>'
```

It seems that AR changed again how the ranges are handled: https://github.com/rails/rails/commit/fed7d7cd7c6e71c1b78a4a2fda7d069d4aa6f468

**Comment by voxik** [comment]
Please note that I am testing rspec-matchers 2.8.0 should it make any difference.

**Comment by mcmire** [comment]
Okay, thanks. I'll throw a fix into the 3.0 release.

**Comment by mcmire** [comment]
Closing this in favor of #762.
